components/EmbedMedia/EmbedMedia.tsx

```diff
@@ -0,0 +1,75 @@
+import { useEffect, useState } from 'react';
+import { cnb } from 'cnbuilder';
+import ReactPlayer from 'react-player/lazy';
+import { Container } from '../Container';
+import { WidthBox, type WidthType } from '../WidthBox';
+import { type PaddingType } from '@/utilities/datasource';
+import { type MediaAspectRatioType, mediaAspectRatios } from '@/utilities/datasource';
+import * as styles from './EmbedMedia.styles';
+
+type EmbedMediaProps = React.HTMLAttributes<HTMLDivElement> & {
+  mediaUrl: string;
+  caption?: React.ReactNode;
+  aspectRatio?: MediaAspectRatioType;
+  boundingWidth?: 'site' | 'full';
+  width?: WidthType;
+  spacingTop?: PaddingType;
+  spacingBottom?: PaddingType;
+  isCaptionInset?: boolean;
+};
+
+export const EmbedMedia = ({
+  mediaUrl,
+  caption,
+  aspectRatio,
+  boundingWidth = 'full',
+  width,
+  spacingTop,
+  spacingBottom,
+  isCaptionInset,
+  className,
+  ...props
+}: EmbedMediaProps) => {
+  /**
+   * This is needed to prevent hydration error for the React Player.
+   * https://github.com/cookpete/react-player/issues/1428
+   */
+  const [hasWindow, setHasWindow] = useState(false);
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      setHasWindow(true);
+    }
+  }, []);
```

Comment:
Ran into a hydration error with React Player using Next.js. Using the solution from here:
https://github.com/cookpete/react-player/issues/1428

I also tried the dynamic import method, but I could not get it to play nice with TypeScript - tried various things, but this `hasWindow` method seems to work just as fine.